core/lib/zksync_core/src/proof_data_handler/request_processor.rs

```diff
@@ -105,44 +97,31 @@ impl RequestProcessor {
             .await
             .map_err(RequestProcessorError::ObjectStore)?;
 
-        let (l1_verifier_config, fri_protocol_version_id) = match self.config.protocol_version_loading_mode {
-            ProtocolVersionLoadingMode::FromDb => {
-
-                let header = self
-                .pool
-                .connection()
-                .await
-                .unwrap()
-                .blocks_dal()
-                .get_l1_batch_header(l1_batch_number)
-                .await
-                .unwrap()
-                .expect(&format!("Missing header for {}", l1_batch_number));
-
-            let protocol_version = header.protocol_version.unwrap();
-            // TODO: What invariants have to hold such that protocol version = fri protocol version?
-            let fri_protocol_version = FriProtocolVersionId::from(protocol_version);
-            (self
-                .pool
-                .connection()
-                .await
-                .unwrap()
-                .protocol_versions_dal()
-                .l1_verifier_config_for_version(protocol_version)
-                .await
-                .expect(&format!(
-                    "Missing l1 verifier info for protocol version {protocol_version:?}",
-                )), fri_protocol_version)
-
-            }
-            ProtocolVersionLoadingMode::FromEnvVar => {
-                (self.l1_verifier_config
-                    .expect("l1_verifier_config must be set while running ProtocolVersionLoadingMode::FromEnvVar mode"),
-                    FriProtocolVersionId::try_from(self.config.fri_protocol_version_id)
-                .expect("Invalid FRI protocol version id"))
+        let header = self
+            .pool
+            .connection()
+            .await
+            .unwrap()
+            .blocks_dal()
+            .get_l1_batch_header(l1_batch_number)
+            .await
+            .unwrap()
+            .expect(&format!("Missing header for {}", l1_batch_number));
 
-            }
-        };
+        let protocol_version = header.protocol_version.unwrap();
```

Comment:
IMHO - we should merge fri protocol version and protocol version.

In theory, we might have protocol version changes that are not impacting the circuits, but that's rare enough and probably not worth the added complexity.